frontend/discourse/app/templates/preferences/profile.gjs

```diff
@@ -1,246 +1,396 @@
-        data-setting-name="user-hide-profile"
-      >
-        <PreferenceCheckbox
-          @labelKey="user.hide_profile"
-          @checked={{@controller.model.user_option.hide_profile}}
-          data-setting-name="user-hide-profile"
-          class="pref-hide-profile"
-        />
-      </div>
-    {{/if}}
-
-    {{#if @controller.canChangeBio}}
-      <div class="control-group pref-bio" data-setting-name="user-bio">
-        <label class="control-label">{{i18n "user.bio"}}</label>
-        <div class="controls bio-composer input-xxlarge">
-          <DEditor @value={{@controller.model.bio_raw}} />
-        </div>
-      </div>
-    {{/if}}
-
-    <div class="control-group pref-timezone" data-setting-name="user-timezone">
-      <label class="control-label">{{i18n "user.timezone"}}</label>
-      <TimezoneInput
-        @value={{@controller.model.user_option.timezone}}
-        @onChange={{fn (mut @controller.model.user_option.timezone)}}
-        class="input-xxlarge"
-      />
-      <DButton
-        @icon="globe"
-        @label="user.use_current_timezone"
-        @action={{@controller.useCurrentTimezone}}
-        class="btn-default"
-      />
-    </div>
-
-    {{#if @controller.model.can_change_location}}
-      <div
-        class="control-group pref-location"
-        data-setting-name="user-location"
-      >
-        <label class="control-label" for="edit-location">{{i18n
-            "user.location"
-          }}</label>
-        <div class="controls">
-          <Input
-            @type="text"
-            @value={{@controller.model.location}}
-            class="input-xxlarge"
-            id="edit-location"
-          />
-        </div>
-      </div>
-    {{/if}}
-
-    {{#if @controller.model.can_change_website}}
-      <div class="control-group pref-website" data-setting-name="user-website">
-        <label class="control-label" for="edit-website">{{i18n
-            "user.website"
-          }}</label>
-        <div class="controls">
-          <Input
-            @type="text"
-            @value={{@controller.model.website}}
-            class="input-xxlarge"
-            id="edit-website"
-          />
-        </div>
-      </div>
-    {{/if}}
-  {{/unless}}
-
-  {{#each @controller.userFields as |uf|}}
-    <div class="control-group" data-setting-name="user-user-fields">
-      <UserField @field={{uf.field}} @value={{uf.value}} />
-    </div>
-  {{/each}}
-  <div class="clearfix"></div>
-
-  {{#unless @controller.showEnforcedRequiredFieldsNotice}}
-    {{#if @controller.siteSettings.allow_profile_backgrounds}}
-      {{#if @controller.canUploadProfileHeader}}
-        <div
-          class="control-group pref-profile-bg"
-          data-setting-name="user-profile-bg"
+const validationFor = (field) => {
+  if (!get(field, "required")) {
+    return null;
+  }
+  return get(field, "field_type") === "confirm" ? "accepted" : "required";
+};
+
+class MutableField extends Component {
```

Comment:
I don't understand what this is for, FormKit should already handle this kind of mutability for fields?